crates/version/build.rs

```diff
@@ -0,0 +1,40 @@
+use std::process::Command;
+
+fn main() {
+    // Get git describe output for build version
+    let git_describe = Command::new("git")
+        .args([
+            "describe",
+            "--always",
+            "--dirty=-modified",
+            "--tags",
+            "--match",
+            "[0-9]*",
+        ])
+        .output()
+        .ok()
+        .and_then(|output| String::from_utf8(output.stdout).ok())
+        .unwrap_or_else(|| "unknown".to_string());
+
+    // Get git commit hash
+    let git_commit = Command::new("git")
+        .args(["rev-parse", "--short", "HEAD"])
+        .output()
+        .ok()
+        .and_then(|output| String::from_utf8(output.stdout).ok())
+        .unwrap_or_else(|| "unknown".to_string());
+
+    // Get rustc version
+    let rustc_version = Command::new("rustc")
+        .arg("--version")
+        .output()
+        .ok()
+        .and_then(|output| String::from_utf8(output.stdout).ok())
+        .map(|s| s.split_whitespace().nth(1).unwrap_or("unknown").to_string())
+        .unwrap_or_else(|| "unknown".to_string());
```

Comment:
the binary name isn't guaranteed to be `rustc`, use the [`rustc_version`](https://docs.rs/rustc_version/latest/rustc_version/) crate instead